### PR TITLE
Use `hexutil.Uint` for `blockCount` parameter in `feeHistory` method

### PR DESCRIPTION
**Pull request by lightclient**
The current spec uses an `Integer` type for the `blockCount` parameter. The [schema](https://github.com/ethereum/eth1.0-specs/blob/bdabd69d0943e518d9bdadf1ef253ae3f72368ce/json-rpc/spec.json#L1788-L1793) for this type defines it to be a hex string.

(h/t @ryanschneider for finding)

### CODE REVIEW

**Review by ryanschneider — COMMENTED** [review]
Note that this will be a breaking change from 1.10.5 which was released with the previous behavior. I'm definitely torn on that.

**Comment by juanfranblanco** [comment]
H!i Also note that the "oldestBlock" in the output is also not Hex

File: eth/gasprice/feehistory.go [hunk 1]
```diff
@@ -275,9 +276,9 @@ func (oracle *Oracle) FeeHistory(ctx context.Context, blocks int, lastBlock rpc.
 	for ; blocks > 0; blocks-- {
 		fees := <-results
 		if fees.err != nil {
-			return 0, nil, nil, nil, fees.err
+			return common.Big0, nil, nil, nil, fees.err
 		}
-		i := int(fees.blockNumber - oldestBlock)
+		i := int(fees.blockNumber) - oldestBlock
```
Comment (rjl493456442) [inline eth/gasprice/feehistory.go:281]: I think we can also use the `uint64` for `fees.blockNumber`. And only convert it to `rpc.BlockNumber` when call some backend APIs.

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -81,19 +81,19 @@ func (s *PublicEthereumAPI) MaxPriorityFeePerGas(ctx context.Context) (*hexutil.
 }
 
 type feeHistoryResult struct {
-	OldestBlock  rpc.BlockNumber  `json:"oldestBlock"`
+	OldestBlock  *hexutil.Big     `json:"oldestBlock"`
```
Comment (rjl493456442) [inline internal/ethapi/api.go:84]: Btw, the spec seems wrong. Can you also open a PR to fix the spec? https://github.com/ethereum/eth1.0-specs/blob/bdabd69d0943e518d9bdadf1ef253ae3f72368ce/json-rpc/spec.json#L248

**Review by rjl493456442 — APPROVED** [review]
Just nitpicks.
@zsfelfoldi Please also check it, especially in the spec the `OldestBlock` type is `rpc.BlockNumber` but I think `hexutil.Big` is better since it can always be resolved.

File: internal/ethapi/api.go (hunk 2 above)
Comment (lightclient) [reply internal/ethapi/api.go:84]: Updated: https://github.com/ethereum/eth1.0-specs/pull/279

File: eth/gasprice/feehistory.go [hunk 3]
```diff
@@ -225,9 +226,9 @@ func (oracle *Oracle) FeeHistory(ctx context.Context, blocks int, lastBlock rpc.
 	)
 	pendingBlock, pendingReceipts, lastBlock, blocks, err = oracle.resolveBlockRange(ctx, lastBlock, blocks, maxHistory)
 	if err != nil || blocks == 0 {
-		return 0, nil, nil, nil, err
+		return common.Big0, nil, nil, nil, err
 	}
-	oldestBlock := lastBlock + 1 - rpc.BlockNumber(blocks)
+	oldestBlock := int(lastBlock) + 1 - blocks
```
[the remark this answers is not in the document]
Comment (lightclient) [reply eth/gasprice/feehistory.go]: done

**Review by zsfelfoldi — APPROVED** [review]
LGTM (after fixing a build failure in the tests)

File: internal/ethapi/api.go (hunk 2 above)
Comment (zsfelfoldi) [reply internal/ethapi/api.go:84]: Thanks for fixing the spec too, it is indeed unnecessary to use `rpc.BlockNumber` for the return value.

**Comment by lightclient** [comment]
@zsfelfoldi doh thanks for fixing the test.

**Comment by holiman** [comment]
@zsfelfoldi don't forget to set the milestone when merging a PR

**Review by Jhart223 — APPROVED** [review]

File: internal/ethapi/api.go [hunk 4]
```diff
@@ -87,8 +87,8 @@ type feeHistoryResult struct {
 	GasUsedRatio []float64        `json:"gasUsedRatio"`
 }
 
-func (s *PublicEthereumAPI) FeeHistory(ctx context.Context, blockCount int, lastBlock rpc.BlockNumber, rewardPercentiles []float64) (*feeHistoryResult, error) {
-	oldest, reward, baseFee, gasUsed, err := s.b.FeeHistory(ctx, blockCount, lastBlock, rewardPercentiles)
```
Comment (Farzin3) [inline internal/ethapi/api.go:91]: BTC 1000000.01

**Review by Farzin3 — COMMENTED** [review]
``>>> import hashlib
>>> m = hashlib.sha256()
>>> m.update(b"Nobody inspects")
>>> m.update(b" the spammish repetition")
>>> m.digest()
b'\x03\x1e\xdd}Ae\x15\x93\xc5\xfe\\\x00o\xa5u+7\xfd\xdf\xf7\xbcN\x84:\xa6\xaf\x0c\x95\x0fK\x94\x06'
>>> m.digest_size
32
>>> m.block_size
64
متراکم تر:

>>>
hashlib.sha224(b"Nobody inspects the spammish repetition").hexdigest()
'a4337bc45a8fc544c03f52dc550cd6e1e87021bc896588bd79e901e2'``

**Comment by Farzin3** [comment]
> ">>> وارد کردن hashlib
> 
> > > > m = hashlib.sha256 () 
> > > > m.update (b "کسی بازرسی نمی کند") 
> > > > m.update (b "تکرار هرزنامه") 
> > > > m.digest () 
> > > > b '\ x03 \ x1e \ xdd} Ae \ x15 \ x93 \ xc5 \ xfe \\ x00o \ xa5u+7 \ xfd \ xdf \ xf7 \ xbcN \ x84: \ xa6 \ xaf \ x0c \ x95 \ x0fK \ x94 \ x06 ' 
> > > > m.digest_size 
> > > > 32 
> > > > m.block_size 
> > > > 64 
> > > > متراکم تر:
> 
> > > >
> 
> hashlib.sha224 (b "هیچ کس تکرار هرزنامه را بررسی نمی کند"). hexdigest () 
> 

>  #`'a4337bc45a8fc544c03f52dc550cd6e1e87021bc896588bd79e901e2` ""

**Pull request merged**